pywemo/subscribe.py

```diff
@@ -219,6 +219,16 @@ def path(self) -> str:
         """Path for the callback to disambiguate multiple subscriptions."""
         return f'/sub/{self.service_name}'
 
+    @property
+    def callback_url(self) -> str:
+        """URL that the device will use to send push notifications."""
+        pywemo_callback_address = os.getenv('PYWEMO_CALLBACK_ADDRESS')
+        if pywemo_callback_address is not None:
+            return f'http://{pywemo_callback_address}{self.path}'
+
+        host = get_ip_address(host=self.device.host)
```

Comment:
Let's move the `os.getenv('PYWEMO_CALLBACK_ADDRESS')` into the `get_ip_address` method, and just use `PYWEMO_CALLBACK_ADDRESS` for the IP address only, not the port. That way this works for other callers of `get_ip_address` also.

Reply:
Sounds good, but if we do that, I think I'm going to have to add a third environment variable for the host callback port.

`PYWEMO_HTTP_SERVER_PORT` controls the internal port, so if I start a container with `-p 123:456`, we need a way to pass the host port to the device.

Reply:
Ah! I see. That's an edge case that I didn't account for. Thanks for clarifying.

The callback address & port also needs to be used over in [ssdp.py](https://github.com/pywemo/pywemo/blob/44184ace8e7ba9a75c44337a0b699179221c7215/pywemo/ssdp.py#L478). WDYT about renaming `get_ip_addr` to `get_callback_address`, like this:

```python
def get_callback_address(host: str = '1.2.3.4', port: int) -> str | None:
    """Return the IP address & port to which devices can send push notifications."""
    pywemo_callback_address = os.getenv('PYWEMO_CALLBACK_ADDRESS')
    if pywemo_callback_address is not None:
        return pywemo_callback_address

    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    try:
        sock.connect((host, 9))
        return f'{sock.getsockname()[0]}:{port}'
    except OSError:
        return None
    finally:
        del sock
```

Then, in ssdp.py, I suggest the following changes.

1. In [SSDP_NOTIFY](https://github.com/pywemo/pywemo/blob/44184ace8e7ba9a75c44337a0b699179221c7215/pywemo/ssdp.py#L46) & [SSDP_REPLY](https://github.com/pywemo/pywemo/blob/44184ace8e7ba9a75c44337a0b699179221c7215/pywemo/ssdp.py#L35) change the `http://%s:%d/setup.xml` to `http://%s/setup.xml`
2. Update `send_notify` to call get_callback_address [here](https://github.com/pywemo/pywemo/blob/44184ace8e7ba9a75c44337a0b699179221c7215/pywemo/ssdp.py#L420). Calling it with `get_callback_address(addr, self.callback_port)` will work.
3. Update `respond_to_discovery` to use get_callback_address [here](https://github.com/pywemo/pywemo/blob/44184ace8e7ba9a75c44337a0b699179221c7215/pywemo/ssdp.py#L478). Calling it with `get_callback_address(sock_addr[0], self.callback_port) will work.

Reply:
It took me a while to wrap my mind around this, but I think I've got it working.

Based on this change, I think homeassistant needs to be updated [here](https://github.com/home-assistant/core/blob/1a43f107c492c6d8e531904188a532af1b251cd0/homeassistant/components/wemo/__init__.py#L110) to use `registry.callback_port` instead of `registry.port`.

The other option is to update `SubscriptionRegistry::port` to do what `callback_port` is doing (and then just get rid of callback_port).  I didn't go down this path as it seems like a potentially breaking change - something that relies on `port` being the local port instead of being the port used for callbacks.